debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySQLConnection.java

```diff
@@ -82,4 +88,85 @@ protected static void addDefaults(Configuration.Builder builder) {
     public MySQLConnection(Configuration config) {
         super(config, FACTORY, null, MySQLConnection::addDefaults);
     }
+
+    public DatabaseDifferences databaseAsserts() {
+        if (databaseAsserts == null) {
+            try {
+                final String versionString = connect().queryAndMap("SHOW GLOBAL VARIABLES LIKE 'version'", rs -> {
+                    rs.next();
+                    return rs.getString(2);
+                });
+                if (versionString.startsWith("8.")) {
+                    databaseAsserts = new DatabaseDifferences() {
+                        @Override
+                        public boolean isCurrentDateTimeDefaultGenerated() {
+                            return true;
+                        }
+
+                        @Override
+                        public String currentDateTimeDefaultOptional(String isoString) {
+                            return null;
+                        }
+
+                        @Override
+                        public String geometryDatabaseName() {
+                            return "geometry_test_8";
+                        }
+
+                        /**
+                         * MySQL 8 does not support unknown SRIDs so the case is removed
+                         */
+                        @Override
+                        public int geometryPointTableRecords() {
+                            return 3;
+                        }
+
+                        /**
+                         * MySQL 8 returns X and Y in a different order
```

Comment:
That sounds a bit scary. Is there code in the connector in place which keeps the existing order within emitted messages?

Reply:
Found [this post](https://mysqlserverteam.com/axis-order-in-spatial-reference-systems/) which describes the change a bit. I tend to agree it's fine to pass through the changed order as is. Us abstracting this could be more trouble than gain potentially.